engine/flink/util/src/main/scala/pl/touk/nussknacker/engine/flink/util/exception/RateMeterExceptionConsumer.scala

```diff
@@ -1,26 +1,34 @@
 package pl.touk.nussknacker.engine.flink.util.exception
 
-import cats.data.NonEmptyList
+import pl.touk.nussknacker.engine.api.exception.{EspExceptionInfo, NonTransientException}
 import pl.touk.nussknacker.engine.api.runtimecontext.EngineRuntimeContext
-import pl.touk.nussknacker.engine.api.{JobData, Lifecycle}
 import pl.touk.nussknacker.engine.flink.api.exception.FlinkEspExceptionConsumer
-import pl.touk.nussknacker.engine.flink.util.metrics.WithMetrics
-import pl.touk.nussknacker.engine.util.exception.GenericRateMeterExceptionConsumer
-import pl.touk.nussknacker.engine.util.metrics.{InstantRateMeterWithCount, RateMeter}
+import pl.touk.nussknacker.engine.util.exception.ExceptionRateMeter
+import pl.touk.nussknacker.engine.util.metrics.WithMetrics
 
-class RateMeterExceptionConsumer(val underlying: FlinkEspExceptionConsumer) extends FlinkEspExceptionConsumer with WithMetrics with GenericRateMeterExceptionConsumer with Lifecycle  {
+//For Flink we count metrics in ExceptionHandler, not e.g. in NodeCountListener, so that it's easier to
+//handle errors consistently in CustomStreamTransformers
+class RateMeterExceptionConsumer(val underlying: FlinkEspExceptionConsumer) extends FlinkEspExceptionConsumer with WithMetrics {
+
+  private var exceptionRateMeter: ExceptionRateMeter = _
```

Comment:
Can it be `transient`? Not sure if it needed but looks like it can be needed.

Reply:
Don't think it needs to be, but it won't hurt, added :+1: